### DISCUSSION TITLE
Game Development

### DISCUSSION THREAD

**Discussion by Jaayson0** [practical-tutorials/project-based-learning#658] [Q&A]
**What are the Languages used for game development?
What's the Easiest? 
Where should I start?
Any tips on How should I start?**
I'm still new to programming, I only know the basics of Python and HTML/CSS, I am here to seek some tips and knowledge from you guys☺️

**Comment by Alwin-Sajeev** [comment]
Hey! Learning C# and Unity is a great move for game development. Unity is beginner-friendly and powerful enough for advanced developers too. Brackeys has some fantastic beginner tutorials on YouTube, and there are plenty of other resources as well. You'll get hands-on with scripting in C# and learn Unity's tools step by step. Let me know if you need help with anything specific along the way!

**Comment by abhaypandeyww** [comment]
Python, HTML/CSS

**Comment by AdrianPana** [comment]
Unity, as suggested, is the most popular Game Engine. However you could start with Godot, it's also pretty popular and they have their own language that is similar to Python. It's also beginner friendly. Search up a youtube video that introduces you to the editor, and maybe first steps to make a game. Try to make a circle appear on screen and move it through code, then through WASD. Look into adding a sprite to the circle, maybe an enemy that follows the player position and kills on collision. Think of something you'd add to the game, and try to implement it, if you have no idea, look it up, follow a tutorial, try to reiterate on the idea and make it your own.

**Comment by HashirSaudKhan** [comment]
### Languages for Game Development:
1. C++: Used in game engines like Unreal Engine; highly performant but complex.
2. C#: Popular in Unity; beginner-friendly with a balance of power and simplicity.
3. Python: Great for simple games (e.g., with Pygame), not ideal for heavy 3D games.
4. JavaScript: Used for browser-based games with frameworks like Phaser.
5. Java: Often used in Android games or with libraries like LibGDX.

**Reply by shizodini** [reply]
great

**Comment by DSrinivasKarthik** [comment]
Hey! Common game dev languages are C#, C++, Python, and JavaScript.
C# with Unity is one of the easiest to start with since it has lots of tutorials and is beginner-friendly.
Since you know Python, try Pygame first.
Start small (like pong or a platformer), follow YouTube tutorials, and have fun bro.

**Comment by manishp608** [comment]
You can start your game development using C# which is used in Unity. Also pygame lets us make 2D games, if your workstation allows even unreal is an option. Godot is a free open-source solution if you are looking for FOSS. Though workflows might differ for all of these, I would recommend starting small. I started with making pong.

**Comment by FelixFriju** [comment]
> **What are the Languages used for game development? What's the Easiest? Where should I start? Any tips on How should I start?** I'm still new to programming, I only know the basics of Python and HTML/CSS, I am here to seek some tips and knowledge from you guys☺️

learn Godot it is easy

**Comment by Tuhin092005** [comment]
If you're already comfortable with the basics of Python, you can start with a simple 2D game rather than jumping directly into a complex 3D project.

For languages, some common choices are:

* **Python** — beginner-friendly and good for learning game-development concepts with libraries such as Pygame.
* **C#** — widely used with Unity and a good option if you want to explore more full-featured game development.
* **C++** — commonly used with Unreal Engine, but it has a steeper learning curve for beginners.
* **JavaScript** — useful for browser-based games and works well with HTML/CSS.

Since you already know some Python, I'd suggest starting with **Python + Pygame**. Build something small like Pong, Snake, Tic-Tac-Toe, or a simple platformer. This will help you learn game loops, keyboard input, collision detection, scoring, and basic game logic.

Once you're comfortable with those concepts, you can move to an engine such as Unity and learn C# if you want to build larger games.

The most important tip is to start small and finish a few simple projects instead of trying to build a large game immediately. Good luck! 😊
